src/Storefront/Theme/ScssPhpCompiler.php

```diff
@@ -47,10 +74,116 @@ public function compileString(AbstractCompilerConfiguration $config, string $scs
             $this->compiler->setImportPaths($importPaths);
         }
 
-        $css = $this->compiler->compileString($scss, $path)->getCss();
+        // If path is a valid file path and not a virtual path like 'theme-scss:'
+        if ($path !== null && !str_starts_with($path, 'theme-scss:') && file_exists($path)) {
+            $css = $this->compiler->compileFile($path)->getCss();
+        } else {
+            // For virtual paths or when no path is provided, use compileString
+            $css = $this->compiler->compileString($scss)->getCss();
+        }
 
         $this->reset(); // Reset compiler for multiple usage
 
         return $css;
     }
+
+    private function generateCacheKey(AbstractCompilerConfiguration $config, string $scss, ?string $path): string
+    {
+        $outputStyle = $config->getValue('outputStyle');
+        $importPaths = $config->getValue('importPaths');
+
+        // Create an array of serializable config values
+        $serializableConfig = [
+            'outputStyle' => $this->makeSerializable($outputStyle),
+            'importPaths' => $this->makeSerializable($importPaths),
+        ];
+
+        $configHash = Hasher::hash(serialize($serializableConfig));
+
+        // Add timestamp to cache key to invalidate cache on file changes
+        $fileTimestamp = '';
+        if ($path !== null && file_exists($path)) {
```

Comment:
would be nice, if we could use the Symfony FileSystem component instead of native functions. if we inject it, we could even mock it easily in tests